extensions/cms/src/cmsResource/cmsResourceCommands.ts

```diff
@@ -95,18 +97,24 @@ export function deleteRegisteredServerCommand(appContext: AppContext, tree: CmsR
 		if (!(node instanceof RegisteredServerTreeNode)) {
 			return;
 		}
-		appContext.apiWrapper.showWarningMessage(
-			`${localize('cms.confirmDeleteServer', 'Are you sure you want to delete')} ${node.name}?`,
-			localize('cms.yes', 'Yes'),
-			localize('cms.no', 'No')).then((result) => {
-				if (result && result === localize('cms.yes', 'Yes')) {
-					appContext.cmsUtils.removeRegisteredServer(node.name, node.relativePath, node.ownerUri).then((result) => {
-						if (result) {
-							tree.notifyNodeChanged(node.parent);
-						}
-					});
+		try {
+			let result = await appContext.apiWrapper.showWarningMessage(
+				`${localize('cms.confirmDeleteServer', 'Are you sure you want to delete')} ${node.name}?`,
+				localize('cms.yes', 'Yes'),
+				localize('cms.no', 'No'));
+			if (result && result === localize('cms.yes', 'Yes')) {
+				try {
+					let removeResult = await appContext.cmsUtils.removeRegisteredServer(node.name, node.relativePath, node.ownerUri);
+					if (removeResult) {
+						tree.notifyNodeChanged(node.parent);
+					}
+				} catch (error) {
+					return Promise.reject(new Error(error.message));
 				}
-			});
+			}
+		} catch (error) {
+			return Promise.reject(new Error(error.message));
+		}
```

Comment:
Last time I'm going to call this out - but I don't know why you keep having this catch-and-rethrow. It doesn't add anything and just hides the original information.